templates/nextjs/src/app/page.tsx

```diff
@@ -1,120 +1,83 @@
-      return toast.error("Contract not loaded");
-    }
-
-    if (walletBalance?.eq(0)) {
-      return toast.error(
-        <span>
-          Your wallet does not have enough funds. Please top it up using the{" "}
-          <Link href={FAUCET_LINK} target="_blank">
-            faucet.
-          </Link>
-        </span>,
-      );
-    }
-
-    // Call the increment_counter function on the contract
-    const { waitForResult } = await contract.functions
-      .increment_counter(bn(1))
-      .call();
-
-    // Wait for the transaction to be mined, and then read the value returned
-    const { value } = await waitForResult();
-
-    setCounter(value.toNumber());
-
-    await refreshWalletBalance?.();
-  };
+import { useConnectUI, useIsConnected } from "@fuels/react";
+import { useEffect } from "react";
+
+import { useRouter } from "../hooks/useRouter";
+import Button from "../components/Button";
+import Info from "../components/Info";
+import Wallet from "../components/Wallet";
+import Contract from "../components/Contract";
+import Predicate from "../components/Predicate";
+import Script from "../components/Script";
+import Faucet from "../components/Faucet";
+
+function App() {
+  const { connect } = useConnectUI();
+  const { isConnected, refetch } = useIsConnected();
+  const { view, views, setRoute } = useRouter();
+
+  useEffect(() => {
+    refetch();
+  }, [refetch]);
 
   return (
-    <>
-      <div className="flex gap-4 items-center">
-        <FuelLogo />
-        <h1 className="text-2xl font-semibold ali">Welcome to Fuel</h1>
+    <main
+      data-theme="dark"
+      className="flex items-center justify-center lg:pt-6 dark:text-zinc-50/90"
+    >
+      <div id="container" className="mx-8 mb-32 w-full max-w-6xl">
+        <nav id="nav" className="flex items-center justify-center py-1 md:py-6">
+          <a href="https://fuel.network/" target="_blank" rel="noreferrer">
+            <img src="./logo_white.png" alt="Fuel Logo" className="w-[124px]" />
+          </a>
+        </nav>
+
+        <div className="gradient-border rounded-2xl">
+          <div className="grain rounded-2xl p-1.5 drop-shadow-xl">
+            <div
+              id="grid"
+              className="lg:grid lg:grid-cols-7 lg:grid-rows-1 lg:gap-12"
+            >
+              <Info />
+              <div className="col-span-4">
+                <div className="gradient-border h-full rounded-xl bg-gradient-to-b from-zinc-900 to-zinc-950/80">
+                  {!isConnected && (
+                    <section className="flex h-full flex-col justify-center space-y-6 px-4 py-4">
+                      <Button onClick={() => connect()}>Connect Wallet</Button>
+                    </section>
+                  )}
+
+                  {isConnected && (
+                    <section className="flex h-full flex-col justify-center space-y-6 px-4 py-8">
+                      <div className="flex flex-col sm:flex-row gap-3 overflow-x-scroll overflow-visible">
```

Comment:
Maybe I am missing something but why is the `overflow-x-scroll` needed here? It seems to be adding an odd white space after this tag
<img width="639" alt="Screenshot 2024-09-24 at 11 31 04" src="https://github.com/user-attachments/assets/1c65743c-b321-4aa2-8f01-c01ec5780a03">

Reply:
Removed 👍🏻